### ISSUE TITLE
google benchmark support

### ISSUE THREAD

**Issue by marcpawl** [lefticus/cpp_starter_project#76]
Would be nice to have Google benchmark as one of the optional dependencies

**Comment by ddalcino** [comment]
I agree; a CMake flag that people can set would be a great way to encourage our users to benchmark their code.

Please note that it is trivial to add Google Benchmark as a dependency yourself, since it's hosted on [JFrog Bintray](https://bintray.com/conan/conan-center/benchmark%3A_).  Just add this line, perhaps within a conditional, to `cpp_starter_project/CMakeLists.txt`:

    set(CONAN_EXTRA_REQUIRES ${CONAN_EXTRA_REQUIRES} benchmark/1.5.1@_/_)

An argument could be made that benchmarks should be required rather than optional, and that Google Benchmark actually belongs in `cpp_starter_project/cmake/Conan.cmake` right next to `catch2`, as an equally important part of every project. This is a matter of opinion though.

**Comment by aminya** [comment]
I prefer that we add support for [`Coz`](https://github.com/plasma-umass/coz). It is the best tool for benchmarking and profiling.
